core/EE_System.core.php

```diff
@@ -419,7 +419,7 @@ public function detect_activations_or_upgrades()
         foreach ($this->registry->addons as $addon) {
             if ($addon instanceof EE_Addon) {
                 // detect teh request type for that addon
-                $addon->detect_activation_or_upgrade();
+                $addon->detect_req_type();
```

Comment:
`detect_req_type()` only calls `detect_activation_or_upgrade()` if it hasn't been done already